src/main/java/CommandParser.java

```diff
@@ -0,0 +1,84 @@
+package main.java;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandParser {
+    private String input;
+    private String type;
+
+    public CommandParser(String input) {
+        this.input = input;
+        int firstSpaceIndex = input.indexOf(" ");
+        if (firstSpaceIndex != -1) {
+            this.type = input.substring(0, firstSpaceIndex);
+        } else {
+            this.type = input;
+        }
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public String[] parseInput() throws ChatException {
+        switch (this.type) {
+            case "todo":
+                String regex1 = "^(\\w+)(\\s)(.*)";
+                Pattern pattern1 = Pattern.compile(regex1);
+                Matcher matcher1 = pattern1.matcher(input);
```

Comment:
Like @willowisp01 mentioned about the naming of the variable, avoid using numbers to distinguish cases.

You can try to combine the lines into one
```suggestion
                Matcher todoMatcher = Pattern.compile("^(\\w+)(\\s)(.*)").matcher(input);
```